pilot/pkg/proxy/envoy/v2/util.go

```diff
@@ -115,3 +113,21 @@ func buildProtoStruct(name, value string) *google_protobuf.Struct {
 		},
 	}
 }
+
+// messageToStruct is the most inefficient way to pass a struct, but will do for first
+// iteration.
```

Comment:
use `util.MessageToStruct` and complain to protobuf team what's up with the conversion.
a whole bunch of people need a better way.

Reply:
We duplicated the method for now, again only RDSv2 is using control-plane directly and we'll evaluate when all is done.

Reply:
Please share the "efficient" way with the community. Lots of people complained about this exact problem.

Reply:
Sure, we'll do that.